### DISCUSSION TITLE
Embedding external content

### DISCUSSION THREAD

**Discussion by SilentGlasses** [squidfunk/mkdocs-material#3945] [Q&A] [answered]
When embedding external content, teh docs show how to embed files but unless i missed it, it's not clear if it is possible to embed only a section of a file.

Example:  If we want to highlight a specific section of code from a repo to go with instructions in a documentation page how would I embed filename.yml but only the content between lines 23 and 28?

**Comment by squidfunk** [comment, marked as answer]
In general, using the [Snippets](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/) extension, but embedding specific lines is currently not supported:

- https://github.com/facelessuser/pymdown-extensions/issues/1462

**Reply by facelessuser** [reply]
One of the reasons I've yet to add such support to Snippets is I view the including specific line numbers as a very fragile way to handle such includes. While I haven't ruled it out completely, I won't lie and say I'm a fan of the idea.

**Reply by squidfunk** [reply]
Markers or some other similar anchors would suddenly be more stable, but adding support for lines could be a first good option. Other inclusion methods could be added later on.

**Reply by facelessuser** [reply]
I agree that Markers would be way more stable, like some kind of named section.

**Reply by ssbarnea** [reply]
I tried to use snippets to include examples what are expected to be inside code blocks but it does not work. I am still trying to find documentation about how to make code examples that get the code from outside the markdown file.

**Reply by vinisalazar** [reply]
@ssbarnea same here. Can't get it to work.

**Reply by vinisalazar** [reply]
@ssbarnea after a few attempts I realised it required the full path of the embedded file, not the relative path (in relation to the file where it's being embedded INTO), as noted [in this comment](https://github.com/squidfunk/mkdocs-material/issues/4414#issuecomment-1398337336) #4414
